app/discovery/provider/consulcatalog/consulcatalog.go

```diff
@@ -0,0 +1,95 @@
+package consulcatalog
+
+import (
+	"context"
+	"fmt"
+	"github.com/umputun/reproxy/app/discovery"
+	"log"
+	"sync"
+	"time"
+)
+
+//go:generate moq -out consul_client_mock.go -skip-ensure -fmt goimports . ConsulClient
+
+type ConsulClient interface {
+	Get() ([]*ConsulService, error)
+}
+
+type ConsulCatalog struct {
+	client  ConsulClient
+	events  chan discovery.ProviderID
+	autoAPI bool
```

Comment:
is this even a thing here? I'm not sure if auto makes any sense in consul provider